standards/ICRC-3/README.md

```diff
@@ -0,0 +1,206 @@
+independently without breaking them, it also means that a client may not receive all the information that the server is sending, e.g. in case the client schema lacks some fields that the server schema has.
+
+This loss of information is not an option for `ICRC-3`. The client must receive the same exact data the server sent. For this reason, `ICRC-3` introduces the `Value` type which never changes: 
+
+```
+type Value = variant { 
+    Blob : blob; 
+    Text : text; 
+    Nat : nat; // do we need this or can we just use Int?
+    Int : int;
+    Array : vec Value; 
+    Map : vec record { text; Value }; 
+};
+```
+
+Servers must serve the transaction log as list of `Value` where each `Value` represent a single transaction in the transaction log.
+
+## Value Hash
+
+`ICRC-3` specifies a standard hash function over `Value`.
+
+This hash function should be used by Ledgers to calculate the hash of the parent of a transaction and by clients to verify the downloaded transaction log.
+
+The hash function is the [representation-independent hashing of structured data](https://internetcomputer.org/docs/current/references/ic-interface-spec#hash-of-map) used by the IC:
+- the hash of a `Blob` is the hash of the bytes themselves
+- the hash of a `Text` is the hash of the bytes representing the text
+- the hash of a `Nat` is the [`leb128`](https://en.wikipedia.org/wiki/LEB128#Unsigned_LEB128) encoding of the number
+- the hash of an `Int` is the [`sleb128`](https://en.wikipedia.org/wiki/LEB128#Signed_LEB128) encoding of the number
+- the hash of an `Array` is the hash of the concatenation of the hashes of all the elements of the array
+- the hash of a `Map` is the hash of the concatenation of all the hashed items of the map sorted. A hashed item is the tuple composed by the hash of the key and the hash of the value.
+
+## Interaction with other standards
+
+Each standard that adheres to `ICRC-3` must define the list of transactions types they define and/or extend together with their `Value` schema and the function that converts a [`Value`](#value) to that type. Transaction types are well-typed records that are easy to consume by clients.
+
+`Value`s representing transactions must have the following schema properties:
+1. they must be of type `Map`
+1. they must have a field `tx: Map` which describes the transaction
+1. they must have a field `op: Text` inside `tx` which describes the type of transactions, e.g. "ICRC1_Burn" for `ICRC1_Burn` transactions
+1. all transactions with `index > 0` must have a top-level field called `phash: Blob` which contains the [hash](#value-hash) of the previous transactions. 
```

Comment:
```suggestion
1. each transaction with `index > 0` must have a top-level field called `phash: Blob` which contains the [hash](#value-hash) of the previous transaction. 
```

Reply:
I think we also need to specify what is `phash` for the first transaction since this is how one determines that it's the beginning of the chain. Either it's None, or a value with no hash preimage (maybe the all 0 blob).